### ISSUE TITLE
Resolve symlink before call to `lipo` to compile in Xcode 10

### ISSUE THREAD

**Issue by DarkDust** [CocoaPods/CocoaPods#7814]
* [X] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md). 🌈

# Report

## What did you do?

Tried to build my project in Xcode 10 beta 1 which includes a pod that wraps an existing framework. This framework contains i386 code which is now getting stripped.

## What did you expect to happen?

Correctly run the embed pods phase.

## What happened instead?

The "[CP] Embed Pods Frameworks" phase now fails on a pod that merely wraps a framework with the following error message:

    <path to myframework>: bundle format is ambiguous (could be app or framework)

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
    RubyGems : 2.4.8
        Host : Mac OS X 10.13.5 (17F70a)
       Xcode : 9.3.1 (9E501)
         Git : git version 2.15.1 (Apple Git-101)
Ruby lib dir : /Users/haisenko/.rvm/rubies/ruby-2.2.2/lib
Repositories : eqpodspecs - git@vcs.muc.equinux.net:equinux/eqPodSpecs.git @ e011b702a3bea20d71c595d0cef20b77061f7db0
               equinux-eqpodspecs - ssh://git@vcs.muc.equinux.net/equinux/eqPodSpecs.git @ e011b702a3bea20d71c595d0cef20b77061f7db0
               master - https://github.com/CocoaPods/Specs.git @ 8ef5b361deaaa25607575fa29d9a4a1b41078896
```

**NOTE**: Xcode is actually version 10.0 beta (10L176w)

### Installation Source

```
Executable Path: /Users/haisenko/.rvm/gems/ruby-2.2.2/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

# Analysis

The problem is that `lipo` replaces the symlink to the framework binary with the stripped binary. So with a structure like:

    My.framework/My -> Versions/A/My
    My.framework/Versions/A/My

The call to `lipo` on `My.framework/My` then turns the symlink into the stripped binary (and `My.framework/Versions/A/My` remains unstripped!).

This in turn makes `codesign` complain with the error: `bundle format is ambiguous (could be app or framework)`

I was able to work around this by resolving any symlinks in the binary path before calling `lipo` in the  `Pods-MyTarget-frameworks.sh` script:

      # Strip invalid architectures so "fat" simulator / device frameworks work on device
      if [[ "$(file "$binary")" == *"dynamically linked shared library"* ]]; then
        strip_binary="$(perl -MCwd -e 'print Cwd::abs_path shift' "$binary")"
        strip_invalid_archs "$strip_binary"
      fi

(The `perl` command was lifted from https://stackoverflow.com/a/24572274/400056 ; it resolves any symlinks in the given path and outputs the absolute path without symlinks. No idea whether this is suitable as a solution for CocoaPods.)

**Comment by CodingMarkus** [comment]
Duplicate of #7587 which also has a shell only fix that won't require using Perl.

**Comment by dnkoutso** [comment]
That's correct. I will spend sometime to finalize @CodingMarkus suggestion in #7587.

**Comment by dnkoutso** [comment]
@DarkDust you can manually apply the fix in the scripts to verify it works but a PR is out.

**Comment by DarkDust** [comment]
@dnkoutso Thank you, will do so on Monday, I have a day off today. At first glance, the patch looks fine. (If the symlink points to an absolute path, `binary="${dirname}/$(readlink "${binary}")"` will fail, though; on the other hand, a symlink to an absolute path may not be used at that point anyway.)

**Comment by dnkoutso** [comment]
good point, let's try this out and see I dont believe it will be an absolute path ever for a dynamic framework.

Thanks again for trying it out!

**Comment by DarkDust** [comment]
I just manually applied the patch and I can confirm that it does work correctly for my project. I do get the "Destination binary is symlinked…" message as expected.

**Comment by dnkoutso** [comment]
Thank you very much @DarkDust! I will be merging this soon and will ship with 1.6.0.beta.1

**Comment by dnkoutso** [comment]
@DarkDust this was merged and will ship with 1.6.0. Thanks again for verifying this!
